services/app-web/src/components/ChangeHistory/ChangeHistory.tsx

```diff
@@ -29,19 +32,28 @@ export const ChangeHistory = ({
     const { logAccordionEvent } = useTealium()
     const flattenedRevisions = (): flatRevisions[] => {
         const result: flatRevisions[] = []
-
         const contractSubmissions = contract.packageSubmissions.filter(
             (submission) => {
                 return submission.cause === 'CONTRACT_SUBMISSION'
             }
         )
-
+        const reviewActions = contract.reviewStatusActions?.filter(
+            (action) => action !== undefined && action !== null
+        )
         //Reverse revisions to order from earliest to latest revision. This is to correctly set version for each
         // contract & recontract.
-        const reversedRevisions = [
-            ...contractSubmissions,
-            contract.draftRevision,
-        ].reverse()
+        let reversedRevisions: (
+            | ContractPackageSubmission
+            | ContractRevision
+            | ContractReviewStatusActions
+            | undefined
+            | null
+        )[] = [...contractSubmissions, contract.draftRevision]
+        if (reviewActions) {
+            reversedRevisions = reversedRevisions.concat(...reviewActions)
+        }
+        reversedRevisions.reverse()
```

Comment:
I think you may need to sort everything in this array by ugpdateinformation.updatedAt rather than just reversing it. Once we have un-withdraw/un-approve it will be possible to unlock and resubmit in between review actions.

Reply:
Lower in this file on line in the return we sort. So I'm not even sure if the `.reverse` is necessary

```
result.sort(
            (a, b) =>
                new Date(b.updatedAt).getTime() -
                new Date(a.updatedAt).getTime()
```